client/src/util/fetchFromAPI.js

```diff
@@ -13,17 +13,14 @@ export const queryAPI = async (options) => {
     throw new Error("onError is not provided");
   }
   const result = {
-    data: null,
-    error: null,
+    data: "",
+    error: "",
   };
   const response = await fetch(url, method);
   if (response.ok) {
-    result.data = await response.json();
-    //make sure onSuccess is there
-    onSuccess(result);
+    return (result.data = await response.json());
   } else {
     result.error = await response.text();
-    onError(result);
+    toast.error(`Food item not saved.  Error: ${result.error}`);
```

Comment:
What are your thoughts on only letting the server error come through here and removing any references to a domain object (e.g. `Food`) or an interaction type (e.g. `saving`)

Doing so will ensure this function is useful in more scenarios.

Reply:
I agree